### ISSUE TITLE
[JAX GPU] CUDA 12 compiler abort in Triton GEMM fusion autotuner: "Can not combine dim orders and requirements"

### ISSUE THREAD

**Issue by jasminetrail** [jax-ml/jax#37955]
### Description

XLA GPU compilation aborts on CUDA 12 while compiling a fused Triton GEMM generated from a JAX gradient computation involving jnp.linalg.cholesky.  The crash occurs during GEMM fusion autotuning inside:
```
xla/service/gpu/autotuning/gemm_fusion_autotuner.cc
```
with the following internal error:
```
FAILED_PRECONDITION: Can not combine dim orders and requirements.
```
followed by:
```
Aborted (core dumped)
```
The failure is triggered while compiling a fused dot computation that contains multiple layout-changing operations (bitcast, transpose) originating from the transpose/JVP rule of Cholesky during reverse-mode autodiff.

Reproduction cases:
jax[cuda12] → crash
jax[cuda13] → fine

Minimal code to reproduce:
```python
import jax
import jax.numpy as jnp

_key = jax.random.PRNGKey(6)
_key, _k_t1859 = jax.random.split(_key)
t1859 = jax.random.normal(_k_t1859, (16, 12, 7))

def model_6(t1859):
    t1861 = jnp.roll(t1859, shift=1, axis=1)
    t1862 = jnp.nanprod(jnp.clip(t1861, -2.0, 2.0), axis=2)
    t1870 = jax.nn.gelu(t1862)
    _cho_t_t1872 = t1870 / (jnp.linalg.norm(t1870) + 1.0)
    _cho_A_t1872 = jnp.matmul(_cho_t_t1872, jnp.swapaxes(_cho_t_t1872, 0, 1)) + jnp.eye(16)
    t1872 = jnp.linalg.cholesky(_cho_A_t1872)
    t1876 = jax.nn.glu(t1870, axis=-1)
    t1878 = jnp.matmul(t1872, t1876)
    return t1878

_args = (t1859,)

# eager forward 
_eager_out = model_6(*_args)
assert jnp.isfinite(_eager_out).all(), 'eager output not finite'

# jit forward 
_jit_fn = jax.jit(model_6)
_jit_out = _jit_fn(*_args)
assert jnp.isfinite(_jit_out).all(), 'jit output not finite'

# eager/jit output agreement
_out_max = float(jnp.maximum(jnp.abs(_eager_out).max(), jnp.abs(_jit_out).max()))
assert _out_max > 1e4 or jnp.allclose(_eager_out, _jit_out, atol=1e-3, rtol=1e-3), (
    f'eager/jit mismatch: max_diff={jnp.abs(_eager_out - _jit_out).max().item():.6f}'
)

# jit determinism 
_jit_out2 = _jit_fn(*_args)
assert jnp.array_equal(_jit_out, _jit_out2), (
    f'jit non-deterministic: max_diff={float(jnp.abs(_jit_out.astype(jnp.float32) - _jit_out2.astype(jnp.float32)).max()):.6f}'
)

# vmap with batch-1
try:
    _vmap_args = tuple(a[None] for a in _args)
    _vmap_out = jax.vmap(model_6)(*_vmap_args)[0]
    _vmap_max = float(jnp.maximum(jnp.abs(_eager_out).max(), jnp.abs(_vmap_out).max()))
    assert _vmap_max > 1e4 or jnp.allclose(_eager_out, _vmap_out, atol=1e-4, rtol=1e-4), (
        f'vmap/eager mismatch: max_diff={jnp.abs(_eager_out - _vmap_out).max().item():.6f}'
    )
except Exception as _vmap_err:
    raise AssertionError(f'vmap failed: {_vmap_err}')

# jax.grad
_grad_fn = jax.grad(lambda *_a: model_6(*_a).sum(), argnums=(0,))
_grads = _grad_fn(*_args)
assert all(jnp.isfinite(g).all() for g in _grads), 'non-finite gradients'

# jit(grad) vs eager grad agreement
_jit_grad_fn = jax.jit(jax.grad(lambda *_a: model_6(*_a).sum(), argnums=(0,)))
_jit_grads = _jit_grad_fn(*_args)
assert all(jnp.isfinite(g).all() for g in _jit_grads), 'non-finite jit gradients'
_grad_max = max(
    float(jnp.maximum(jnp.abs(_a).max(), jnp.abs(_b).max()))
    for _a, _b in zip(_grads, _jit_grads)
)
assert _grad_max > 1e4 or all(
    jnp.allclose(_a, _b, atol=1e-3, rtol=1e-3)
    for _a, _b in zip(_grads, _jit_grads)
), (
    f'eager/jit grad mismatch: max_diff={max(jnp.abs(_a - _b).max().item() for _a, _b in zip(_grads, _jit_grads)):.6f}'
)
print("success")
```

Error logs:
2026-05-20 02:01:52.376279: F external/xla/xla/service/gpu/autotuning/gemm_fusion_autotuner.cc:1136] Non-OK-status: executable.status()
Status: FAILED_PRECONDITION: Can not combine dim orders and requirements. - Failure occured when compiling fusion gemm_fusion_dot.226 with config '{block_m:16,block_n:16,block_k:16,split_k:1,num_stages:1,num_warps:2,num_ctas:1}'
Fused HLO computation:
%gemm_fusion_dot.226_computation (parameter_0.4: f32[16,16], parameter_1.3: f32[16,16], parameter_2.3: f32[16,12,1], parameter_3.3: f32[16,12,1], parameter_4.2: f32[16,12], parameter_5.2: f32[]) -> f32[16,12] {
  %parameter_0.4 = f32[16,16]{0,1} parameter(0)
  %bitcast.99 = f32[16,16]{1,0} bitcast(%parameter_0.4)
  %constant.45 = f32[] constant(0.5)
  %broadcast.62 = f32[16,16]{1,0} broadcast(%constant.45), dimensions={}
  %multiply.52 = f32[16,16]{1,0} multiply(%bitcast.99, %broadcast.62), metadata={op_name="jit(<lambda>)/jit(main)/transpose(jvp(jit(cholesky)))/div" source_file="/home/test/programs/program_6.py" source_line=14}
  %parameter_1.3 = f32[16,16]{0,1} parameter(1)
  %bitcast.100 = f32[16,16]{1,0} bitcast(%parameter_1.3)
  %multiply.53 = f32[16,16]{1,0} multiply(%bitcast.100, %broadcast.62), metadata={op_name="jit(<lambda>)/jit(main)/transpose(jvp(jit(cholesky)))/div" source_file="/home/test/programs/program_6.py" source_line=14}
  %transpose.22 = f32[16,16]{1,0} transpose(%multiply.53), dimensions={1,0}, metadata={op_name="jit(<lambda>)/jit(main)/transpose(jvp(jit(cholesky)))/transpose" source_file="/home/test/programs/program_6.py" source_line=14}
  %add.21 = f32[16,16]{1,0} add(%multiply.52, %transpose.22), metadata={op_name="jit(<lambda>)/jit(main)/transpose(jvp(jit(cholesky)))/add_any" source_file="/home/test/programs/program_6.py" source_line=14}
  %bitcast.101 = f32[16,16]{0,1} bitcast(%add.21)
  %parameter_2.3 = f32[16,12,1]{2,1,0} parameter(2)
  %parameter_3.3 = f32[16,12,1]{2,1,0} parameter(3)
  %multiply.54 = f32[16,12,1]{2,1,0} multiply(%parameter_2.3, %parameter_3.3), metadata={op_name="jit(<lambda>)/jit(main)/jvp(jit(nanprod))/mul" source_file="/home/test/programs/program_6.py" source_line=10}
  %bitcast.102 = f32[16,12]{1,0} bitcast(%multiply.54), metadata={op_name="jit(<lambda>)/jit(main)/jvp(jit(nanprod))/mul" source_file="/home/test/programs/program_6.py" source_line=10}
  %parameter_4.2 = f32[16,12]{1,0} parameter(4)
  %broadcast.64 = f32[16,12]{1,0} broadcast(%constant.45), dimensions={}
  %multiply.55 = f32[16,12]{1,0} multiply(%parameter_4.2, %broadcast.64), metadata={op_name="jit(<lambda>)/jit(main)/mul" source_file="/home/test/programs/program_6.py" source_line=11}
  %multiply.56 = f32[16,12]{1,0} multiply(%bitcast.102, %multiply.55), metadata={op_name="jit(<lambda>)/jit(main)/mul" source_file="/home/test/programs/program_6.py" source_line=11}
  %parameter_5.2 = f32[] parameter(5)
  %sqrt.4 = f32[] sqrt(%parameter_5.2), metadata={op_name="jit(<lambda>)/jit(main)/jvp(jit(norm))/sqrt" source_file="/home/test/programs/program_6.py" source_line=12}
  %constant.47 = f32[] constant(1)
  %add.22 = f32[] add(%sqrt.4, %constant.47), metadata={op_name="jit(<lambda>)/jit(main)/add" source_file="/home/test/programs/program_6.py" source_line=12}
  %broadcast.65 = f32[16,12]{1,0} broadcast(%add.22), dimensions={}, metadata={op_name="jit(<lambda>)/jit(main)/div" source_file="/home/test/programs/program_6.py" source_line=12}
  %divide.11 = f32[16,12]{1,0} divide(%multiply.56, %broadcast.65), metadata={op_name="jit(<lambda>)/jit(main)/div" source_file="/home/test/programs/program_6.py" source_line=12}
  ROOT %dot.9 = f32[16,12]{1,0} dot(%bitcast.101, %divide.11), lhs_contracting_dims={0}, rhs_contracting_dims={0}, metadata={op_name="jit(<lambda>)/jit(main)/dot_general" source_file="/home/test/programs/program_6.py" source_line=13}
}
Aborted (core dumped)

### System info (python version, jaxlib version, accelerator, etc.)

jax:    0.10.1
jaxlib: 0.10.1
numpy:  2.4.4
python: 3.13.5
